cpp/include/cudf/functions.h

```diff
@@ -927,6 +928,61 @@ gdf_error gdf_quantile_aprrox(	gdf_column*  col_in,       //input column with 0
                                 void*        t_erased_res, //type-erased result of same type as column;
                                 gdf_context* ctxt);        //context info
 
+
+/* --------------------------------------------------------------------------*/
+  /**
+   * @brief Sorts a set of columns based on specified "key" columns. Returns a column containing
+   * the offset to the start of each group
+   *
+   * @Param[in]  num_data_cols         The number of columns in the dataset
+   * @Param[in]  data_cols_in          The input columns in the dataset
+   * @Param[in]  num_key_cols      The number of key columns
+   * @Param[in]  key_col_indices   The column indices of the input dataset that are they keys by which the data will be grouped by
+   * @Param[out] data_cols_out         The dataset sorted by the group by columns (needs to be pre-allocated. Should be the same number of columns and rows as data_cols_in)
+   * @Param[out] group_start_indices   An array containing the starting indices of each key (group). (needs to be pre-allocated to have the same number of rows as data_cols_in)
+   * @Param[out] num_group_start_indices   The number of elements in group_start_indices
+   * @Param[in]  ctxt                  The context used to control how nulls are treated in a sort and in group by
+   *   context->flag_nulls_sort_behavior
+   *    GDF_NULL_AS_LARGEST = Nulls are are treated as largest, 
+   *    GDF_NULL_AS_SMALLEST = Nulls are treated as smallest, 
+   *    GDF_NULL_AS_LARGEST_FOR_MULTISORT = Special multi-sort case any row with null is largest>
+   *   context-> flag_groupby_include_nulls 
+   *    0 = Nulls are ignored in group by keys (Pandas style),
+   *    1 = Nulls are treated as values in group by keys where NULL == NULL (SQL style)>
+   *
+   * @Returns gdf_error with error code on failure, otherwise GDF_SUCESS
+   */
+  /* ----------------------------------------------------------------------------*/
+gdf_error gdf_group_by_without_aggregations(gdf_size_type num_data_cols,
+                                            gdf_column** data_cols_in,
```

Comment:
```suggestion
                                            gdf_column const * const * data_cols_in,
```

Reply:
I agree that this should be made const, but this would require changing all the underlying sort functions and utility functions it uses. This would require a cascade of changes all over the codebase (order by, group by, joins, hash generation, windowed functions, ...) including some stuff that is currently being worked on. If we want to do this, I would suggest doing this as part of a separate PR

Reply:
Understood. Makes sense as a follow-on PR. The `const` correctness of the API and internals is certainly lax.